sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTest.cs

```diff
@@ -90,70 +91,52 @@ public void ThreadCGUD_MemberAUR_MessageGSU_NotificationT()
-            ChatMessage message5 = chatThreadClient3.GetMessage(messageId5);
-            ChatMessage message6 = chatThreadClient3.GetMessage(messageId6);
+
+            var messagesMapping = new Dictionary<int, (string, ChatMessage)>
+            {
+                { 1, (message.Id, message) },
+                { 2, ExecuteSendGetMessage(2, chatThreadClient2, messageTemplate, displayNameTemplate, ChatMessagePriority.High) }
+            };
+            for (int i = 3; i < 7; i++)
+            {
+                messagesMapping.Add(i, ExecuteSendGetMessage(i, chatThreadClient3, messageTemplate, displayNameTemplate, ChatMessagePriority.High));
+            }
 
             #region Snippet:Azure_Communication_Chat_Tests_E2E_GetMessages
             Pageable<ChatMessage> messages = chatThreadClient.GetMessages();
             #endregion Snippet:Azure_Communication_Chat_Tests_E2E_GetMessages
-            Pageable<ChatMessage> messages2 = chatThreadClient2.GetMessages();
-            var getMessagesCount = messages.Count();
-            var getMessagesCount2 = messages2.Count();
+            Pageable<ChatMessage> messagesThread2 = chatThreadClient2.GetMessages();
+            var getMessagesCountThread1 = messages.Count();
+            var getMessagesCountThread2 = messagesThread2.Count();
 
             # region Pagination assertions
             Pageable<ChatMessage> messagesPaginationTest = chatThreadClient.GetMessages();
             string? continuationToken = null;
             var expectedPageSize = 2;
-            var messagesCounterTotal = 0;
+            var messagesTotal = 0;
             var messagesCounter = 0;
-            foreach (Page<ChatMessage> messagesPage in messagesPaginationTest.AsPages(continuationToken,2))
+            ulong previousId = ulong.MaxValue;
+            foreach (Page<ChatMessage> messagesPage in messagesPaginationTest.AsPages(continuationToken, expectedPageSize))
             {
-                messagesCounter = 0;
-                foreach (ChatMessage messagePage in messagesPage.Values)
-                {
-                    messagesCounterTotal++;
-                    messagesCounter++;
-                }
-                continuationToken = messagesPage.ContinuationToken;
-                //Last request does not return items
-                if (messagesPage.Values.Count == 0)
-                {
-                    Assert.IsNull(continuationToken);
-                }
-                else
+                (messagesCounter, continuationToken) = AssertPageableChatMessages(messagesPage, expectedPageSize, messagesMapping);
+                messagesTotal+= messagesCounter;
+
+                //Make sure expected order is ok
+                foreach (ChatMessage chatMessage in messagesPage.Values)
                 {
-                    Assert.AreEqual(expectedPageSize, messagesCounter);
+                    Assert.Less(ulong.Parse(chatMessage.Id), previousId);
```

Comment:
We should make assumption on the order of the ids. The order should be based on the order we send, right?

Reply:
The order of sending and receiving are not the same, they are actually reverted,  so I wanted to validate 2 things :
1. That what we send is what we receive in the right order (will try to do it more explicit on the next commit and add some comments to help clarifying)
2. That the order on the pagination logic is always kept intact. The conversation service  is sorting messages by default using createdOn and then by Id. Both are ulong that represent an epoc time. I wanted to cover some basic checks to make sure the sdk is not changing that order/behaviour.